apache/superset
# feat: Custom server page length

### SUMMARY

This contribution delivers usability enhancements to the Custom Server Page Length feature in Superset. This upgrade now enables custom input values to persist in the drop-down list. The previous version of this feature posed an inconvenience as any new custom value entered by the user would disappear from the drop-down list when the user moved to a different choice.

The new implementation, however, ensures any custom value entered by the user remains in the drop-down list, thereby enriching user accessibility for frequent inputs. Moreover, the system now auto-arranges the drop-down list every time a new value is included, facilitating easy and efficient navigation.

Additional modifications include introducing a cap on the maximum custom page length value. The cap will be equivalent to either the total rows of data results or a fixed 10k rows in scenarios where the total rows of data exceed 10,000. 

### BEFORE SCREENSHOTS OR ANIMATED GIF

https://github.com/apache/superset/assets/87610632/d6462d1f-1a74-4505-89f2-bf8f459147c5


### AFTER SCREENSHOTS OR ANIMATED GIF

https://github.com/apache/superset/assets/87610632/dbb1d509-5886-4452-a559-4bd465bb6e30


### TESTING INSTRUCTIONS

1. Navigate to the Charts page
2. Choose any dataset
3. Choose table chart
4. Check the Server Pagination checkbox
5. Enter a new value in the selection drop-down list
6. Verify that the new input is sticky in the drop-down list

### ADDITIONAL INFORMATION
